planner_cspace/src/planner_3d.cpp

```diff
@@ -1305,13 +1306,21 @@ class Planner3dNode
     goal_tolerance_ang_ = std::lround(goal_tolerance_ang_f_ / map_info_.angular_resolution);
     cc_.angle_resolution_aspect_ = 2.0 / tanf(map_info_.angular_resolution);
 
-    model_.reset(
-        new GridAstarModel3D(
-            map_info_,
-            ec_,
-            local_range_,
-            cost_estim_cache_.gridmap(), cm_, cm_hyst_, cm_rough_,
-            cc_, range_, path_interpolation_resolution_, grid_enumeration_resolution_));
+    const bool reset_required = force_reset || (previous_range != range_);
```

Comment:
I think the change of the `map_info_` is better to be checked here instead of passing as an argument to prevent distributing the logic.

Reply:
But doing this might need larger change.
(pass new `map_info`, parameters as arguments of `resetGridAstarModel` and skip calling `GridAstarModel3D::updateCostParameters` when unnecessary)
So, it looks ok to merge it as is for now.

Reply:
https://github.com/at-wat/neonavigation/issues/752